toolchain/check/handle_class.cpp

```diff
@@ -696,6 +700,46 @@ static auto CheckCompleteClassType(Context& context, Parse::NodeId node_id,
         {.name_id = SemIR::NameId::Base, .type_id = base_type_id});
   }
 
+  if (class_info.is_dynamic) {
+    llvm::SmallVector<SemIR::InstId> vtable;
+    if (!defining_vptr) {
+      LoadImportRef(context, base_class_info->vtable_id);
+      auto vtable_id = context.constant_values().GetConstantInstId(
+          base_class_info->vtable_id);
+      auto base_vtable_inst_block = context.inst_blocks().Get(
+          context.insts().GetAs<SemIR::Vtable>(vtable_id).virtual_functions_id);
+      for (auto fn_decl_id : base_vtable_inst_block) {
+        auto fn_decl = GetCalleeFunction(context.sem_ir(), fn_decl_id);
+        auto fn = context.functions().Get(fn_decl.function_id);
+        for (auto override_fn_decl_id :
+             context.vtable_stack().PeekCurrentBlockContents()) {
+          auto override_fn_decl =
+              context.insts().GetAs<SemIR::FunctionDecl>(override_fn_decl_id);
+          auto override_fn =
+              context.functions().Get(override_fn_decl.function_id);
+          if (override_fn.virtual_modifier ==
+                  SemIR::FunctionFields::VirtualModifier::Impl &&
+              override_fn.name_id == fn.name_id) {
+            fn_decl_id = override_fn_decl_id;
+          }
+        }
+        vtable.push_back(fn_decl_id);
```

Comment:
Please add a TODO to check the signatures match.

Reply:
Ah, yep - added that in 63c943575200906a0d757e56161b7fc51b5f42c6 - actually looking forward to/was planning to add that in follow-up.